Feat: ConstantSourceNode (#707)
software-mansion/react-native-audio-api: packages/audiodocs/docs/sources/constant-source-node.mdx

---
sidebar_position: 6
---

import AudioNodePropsTable from "@site/src/components/AudioNodePropsTable"
import { Optional, ReadOnly } from '@site/src/components/Badges';

> import InteractivePlayground from '@site/src/components/InteractivePlayground';
> import { useOscillatorPlayground } from '@site/src/components/InteractivePlayground/OscillatorExample/useOscilatorPlayground';
> import SimplifiedOscillator from '@site/src/components/InteractivePlayground/ConstantSourceExample/ConstantSourceExample'

Reviewer: I think you can remove these imports
```suggestion

```

# ConstantSourceNode

The `ConstantSourceNode` is an [`AudioScheduledSourceNode`](/docs/sources/audio-scheduled-source-node) which represents an audio source, that outputs a single constant value.
The `offset` parameter controls this value. Although the node is called "constant" its `offset` value can be automated to change over time, which makes it powerful tool
for controlling multiple other [`AudioParam`](/docs/core/audio-param) values in an audio graph.

> Similar to all of `AudioScheduledSourceNodes`, it can be started only once.

Reviewer 2: ```suggestion
Just like `AudioScheduledSourceNode`, it can be started only once.
```
